celery/backends/database/__init__.py

```diff
@@ -98,11 +99,31 @@ def __init__(self, dburi=None, engine_options=None, url=None, **kwargs):
                 'Missing connection string! Do you have the'
                 ' database_url setting set to a real value?')
 
+        self.session_manager = SessionManager()
+
+        create_tables_at_setup = conf.get("database_create_tables_at_setup", None)
```

Comment:
Shouldn’t there be a new conf in https://github.com/celery/celery/blob/ed546cde241ed2bf236043f0e904482b1f6fa2df/celery/app/defaults.py#L242 ?

Reply:
I understand it works because if it doesn’t exist it defaults to `False`.
But the better practice would be to add a new conf, set it explicitly to `False` in `defaults.py` and then use it here without `get()`.

Reply:
of course, it should definitely be here. I tried to search for places where configuration were defined by running a search against `database_short_lived_sessions` but now that I see how the namespace is defined, I understand why there was no match :)

I'll do the required change

Reply:
How can I issue the warning now that the option is cast with type bool (in https://github.com/celery/celery/blob/ed546cde241ed2bf236043f0e904482b1f6fa2df/celery/app/defaults.py#L58-L59) ? I cannot use default value=None, nor can I use unions.

Any tips ?

Reply:
How can I issue the warning now that the option is cast with type bool (in https://github.com/celery/celery/blob/ed546cde241ed2bf236043f0e904482b1f6fa2df/celery/app/defaults.py#L58-L59) ? I cannot use default value=None because its falsy, nor can I use type unions because they are not supported by Option class.

Any tips ?

I could raise the warning every time the option is set to False, but that seems quite extreme

Reply:
Let's see.. hmm, why do we even want this warning?
Yeah, we can define v6.0.0 will change defaults, but do we really care to warn the user about it, potentially every time?

Reply:
- setting this option to False does nothing more compared to previous versions
- we believe the default value should be True so we need to transition from False to True
- to transition there are two ways I see :
  - issue a warning stating the default will change in a future version
  - change the default value to True in this branch and state the change in the change log. Setting this option to True probably has a low impact for users so it is a reasonable way to do it.

Do you suggest we should follow the second way?

Reply:
> change the default value to True in this branch and state the change in the change log. Setting this option to True probably has a low impact for users so it is a reasonable way to do it.

Makes sense.
You may proceed with the changes to the PR then.

Thanks ! @MarcBresson